packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -305,6 +309,26 @@ export function useConnections({
   } = state;
   const connectingConnectionAttempt = useRef<ConnectionAttempt>();
 
+  useEffect(() => {
+    function reloadConnections() {
+      void loadConnections(dispatch, connectionRepository, {
+        persistOIDCTokens,
+      });
+    }
+
+    connectionStorage.events?.on(
```

Comment:
True, this optional chaining is not necessary right now. I was wondering if these events would be optional whenever we implement a new ConnectionStorage for web, that's why I made it optional, in case they are not implemented it can still work.

Reply:
If you want to make them optional, you should type them as so to guarantee optionalness beyond you knowing right now that you assume that they can be optional 🙂 Usually for cases like this we separate the interface from the implementation, like DataService for example